controller/behavior/denormalizeable.php

```diff
@@ -0,0 +1,188 @@
+<?php
+/**
+ * @package     LOGman
+ * @copyright   Copyright (C) 2011 - 2016 Timble CVBA. (http://www.timble.net)
+ * @license     GNU GPLv3 <http://www.gnu.org/licenses/gpl.html>
+ * @link        http://www.joomlatools.com
+ */
+
+/**
+ * Denormalizeable Controller Behavior
+ *
+ * Allows for de-normalization of activities entries
+ *
+ * @author  Arunas Mazeika <https://github.com/amazeika>
+ * @package Joomlatools\Component\LOGman
+ */
+class ComActivitiesControllerBehaviorDenormalizeable extends KControllerBehaviorAbstract
+{
+    /**
+     * The resources controller
+     *
+     * @var KModelInterface
+     */
+    protected $_controller;
+
+    public function __construct(KObjectConfig $config)
+    {
+        parent::__construct($config);
+
+        $this->_controller = $config->controller;
+    }
+
+    protected function _initialize(KObjectConfig $config)
+    {
+        $config->append(array(
+            'controller' => 'resource'
+        ));
+
+        if ($this->getIdentifier()->getPackage() != 'activities')
```

Comment:
This check is for creating an alias to the resource controller when extending the behavior. When extending the behavior checks if a resource controller exists in the component where the behavior is being used, if not, then an alias to fallback to activities is created. Pretty much what we do in the activity controller.